worker/instancemutater/mutater_test.go

```diff
@@ -5,39 +5,268 @@ package instancemutater_test
 
 import (
 	"github.com/golang/mock/gomock"
+	"github.com/juju/errors"
+	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 	"gopkg.in/juju/names.v2"
 
-	apimocks "github.com/juju/juju/api/instancemutater/mocks"
-	jujutesting "github.com/juju/juju/testing"
+	apiinstancemutater "github.com/juju/juju/api/instancemutater"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/lxdprofile"
+	"github.com/juju/juju/core/status"
+	"github.com/juju/juju/worker/instancemutater"
 	"github.com/juju/juju/worker/instancemutater/mocks"
 )
 
 type mutaterSuite struct {
-	jujutesting.BaseSuite
+	loggerSuite
 
-	tag names.MachineTag
+	tag    names.MachineTag
+	instId string
 
-	logger  *mocks.MockLogger
-	machine *apimocks.MockMutaterMachine
+	facade  *mocks.MockInstanceMutaterAPI
+	machine *mocks.MockMutaterMachine
+	broker  *mocks.MockLXDProfiler
+
+	mutaterMachine *instancemutater.MutaterMachine
 }
 
 var _ = gc.Suite(&mutaterSuite{})
 
 func (s *mutaterSuite) SetUpTest(c *gc.C) {
 	s.tag = names.NewMachineTag("2")
-	s.BaseSuite.SetUpTest(c)
+	s.instId = "juju-23413-2"
+	s.IsolationSuite.SetUpTest(c)
+}
+
+func (s *mutaterSuite) TestProcessMachineProfileChanges(c *gc.C) {
+	defer s.setUpMocks(c).Finish()
+
+	startingProfiles := []string{"default", "juju-testme"}
+	finishingProfiles := append(startingProfiles, "juju-testme-lxd-profile-1")
+
+	s.ignoreLogging(c)()
```

Comment:
Direct expectations are much nicer. Shame we have to share this one with the old technique :)